lib/containers/urls.pm

```diff
@@ -342,4 +299,27 @@ sub get_3rd_party_images {
     return (\@images);
 }
 
+# Return url of specified image as a string. Die when nothing to return.
+# Specify released => 0 or CONTAINERS_UNTESTED_IMAGES=1 for unreleased image.
+sub get_image_url {
+    my (%args) = @_;
+    $args{version} //= get_required_var('VERSION');
+    $args{arch} //= get_required_var('ARCH');
+    $args{distri} //= get_required_var('DISTRI');
+    $args{released} //= get_var('CONTAINERS_UNTESTED_IMAGES') ? 0 : 1;
+    $args{die} //= 1;    # 0 => do not die
+
+    $args{version} =~ s/^Staging:(?<letter>.)$/Tumbleweed/ if is_tumbleweed || is_microos("Tumbleweed");
+
+    my $url = trim(get_var('CONTAINER_IMAGE_TO_TEST'));
```

Comment:
The `trim()` is there to trim spaces at the beginning or end of the variable.
E.g. if you specify `CONTAINER_IMAGE_TO_TEST="registry.suse.de/sle:15.3 "` it will *trim* the last space.

This was suggested by @ilausuch - Ivan, can you help here? :smile:

Reply:
Spaces don't harm. But the trim will just simply fail if the variable is empty, which is wrong approach.
If we want to do checks, I would do it in the right way, using `URI` package from perl, e.g. https://stackoverflow.com/questions/2599047/how-can-i-validate-a-website-url-in-perl
But that's optional, that can be improved later, I would simple go for the code I suggested.

Reply:
I see your point. But I wonder how is then possible that my example worker in all the verification runs both with and without this variable.

Please re-check.

Reply:
Yes. I removed the `trim` - is it resolved?

Reply:
I implemented Ivan's solution, thank you!
```
$url = get_var('CONTAINER_IMAGE_TO_TEST');
return trim($url) if ($url)
```

Reply:
Sorry for insisting, but I still don't understand why we need trim here :) 
Just to remove any spaces in the variable is not a good reason, we would need to do this for every variable, which doesn't make sense.